lotus/sem_ops/sem_agg.py

```diff
@@ -14,6 +15,7 @@ def sem_agg(
     partition_ids: list[int],
     safe_mode: bool = False,
     progress_bar_desc: str = "Aggregating",
+    use_operator_cache: bool = False,
```

Comment:
High level question - should `use_operator_cache` be a parameter here or is it better off in `Settings`? I think there is an argument for the latter, since I see this pattern being used everywhere.

Reply:
putting it in settings is better I'd say